app/screens/channel/channel.js

```diff
@@ -80,6 +87,10 @@ export default class Channel extends PureComponent {
         if (LocalConfig.EnableMobileClientUpgrade && !ClientUpgradeListener) {
             ClientUpgradeListener = require('app/components/client_upgrade_listener').default;
         }
+
+        if (props.showTermsOfService && !props.disableTermsModal) {
```

Comment:
Can you move this to `componentDidMount`? That's where React recommends placing things like this that have side effects